src/linter.ts

```diff
@@ -9,10 +9,12 @@ export interface LinterError {
 }
 
 export default class Linter {
+  private workspaceFolder: any;
   private codeDocument: vscode.TextDocument;
 
   constructor(document: vscode.TextDocument) {
     this.codeDocument = document;
+    this.workspaceFolder = vscode.workspace.workspaceFolders[0];
```

Comment:
`workspaceFolders` doesn't appear to exist in the `workspace` namespace in my environment, though I do see _some_ usages of it in articles. Currently looking into where the gap is.

Reply:
Ah ok, it looks like we need to migrate away from the `vscode` dependency and leverage `@types/vscode` and outlined here https://code.visualstudio.com/api/working-with-extensions/testing-extension#migrating-from-vscode. 

We'll also need a check around if `vscode.workspace.workspaceFolders` is undefined before use to resolve lint error `TS2532`.

I did a quick POC and it now compiles with those changes.

[ ] Add @types/vscode to package.json
[ ] Add mocha to package.json
[ ] Remove vscode from package.json
[ ] Have the types and vscode.engines versions match
[ ] Add check around undefined workspaceFolders for TS2532